src/TestSuite/IntegrationTestCase.php

```diff
@@ -352,17 +384,15 @@ public function delete($url)
      */
     protected function _sendRequest($url, $method, $data = [])
     {
-        $request = $this->_buildRequest($url, $method, $data);
-        $response = new Response();
-        $dispatcher = DispatcherFactory::create();
-        $dispatcher->eventManager()->on(
-            'Dispatcher.invokeController',
-            ['priority' => 999],
-            [$this, 'controllerSpy']
-        );
+        if ($this->_useHttpServer) {
+            $dispatcher = new MiddlewareDispatcher($this);
```

Comment:
Maybe get this from a protected method? So that we can offer the ability to override this object.